File: src/main/java/ru/ioleksiv/telegram/bot/api/annotations/filter/telegram/OrderInfoFilter.java
package ru.ioleksiv.telegram.bot.api.annotations.filter.telegram;

import ru.ioleksiv.telegram.bot.api.annotations.filter.primitive.StringFilter;
import ru.ioleksiv.telegram.bot.api.model.annotation.AnnotationState;
import ru.ioleksiv.telegram.bot.api.model.annotation.CustomValidator;
import ru.ioleksiv.telegram.bot.api.model.annotation.stub.StubOrderInfoValidator;
import ru.ioleksiv.telegram.bot.api.model.objects.payments.OrderInfo;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
@Documented
public @interface OrderInfoFilter {

    AnnotationState status() default AnnotationState.ON;

    String[] validator() default {};

    StringFilter name() default @StringFilter(status = AnnotationState.OFF);

    StringFilter phoneNumber() default @StringFilter(status = AnnotationState.OFF);

    StringFilter email() default @StringFilter(status = AnnotationState.OFF);

    ShippingAddressFilter shippingAddress() default @ShippingAddressFilter(status = AnnotationState.OFF);

}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/annotations/filter/telegram/OrderInfoFilter.kt
package io.github.oleksivio.telegram.bot.api.annotations.filter.telegram

import io.github.oleksivio.telegram.bot.api.annotations.filter.primitive.StringFilter
import io.github.oleksivio.telegram.bot.api.model.annotation.AnnotationState

@Retention
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@MustBeDocumented
annotation class OrderInfoFilter(val status: AnnotationState = AnnotationState.ON, val validator: Array<String> = [], val name: StringFilter = StringFilter(status = AnnotationState.OFF), val phoneNumber: StringFilter = StringFilter(status = AnnotationState.OFF), val email: StringFilter = StringFilter(status = AnnotationState.OFF), val shippingAddress: ShippingAddressFilter = ShippingAddressFilter(status = AnnotationState.OFF))
